### PR TITLE
Remove the CONFIG_DB_INIT flag dependency on db_migrator

### PR DESCRIPTION
**Pull request by vivekrnv**
#### What I did

db_migrator should not depend on CONFIG_DB_INITIALIZED flag. get_hwsku api call busy waits on that flag

#### How I did it

Replace get_hwsku call with get_localhost_info call which takes in a custom config_db object

### CODE REVIEW

File: scripts/db_migrator.py [hunk 1]
```diff
@@ -91,7 +91,9 @@ def __init__(self, namespace, socket=None):
         self.asic_type = version_info.get('asic_type')
         if not self.asic_type:
             log.log_error("ASIC type information not obtained. DB migration will not be reliable")
-        self.hwsku = device_info.get_hwsku()
+
+        # TODO: Update get_hwsku API to take in a custom configDB object
```
Comment (vaibhavhd) [inline scripts/db_migrator.py:95]: Why is this TODO left? How will custom configDB help unblock the busy wait of get_hwsku?

Comment (vivekrnv) [reply scripts/db_migrator.py:95]: When get_hwsku() is used, it calls get_localhost_info and it calls ConfigDBConnector.connect() which busy waits. 

I'll remove the TODO comment

**Review by dgsudharsan — APPROVED** [review]

**Review by vaibhavhd — APPROVED** [review]

**Comment by vaibhavhd** [comment]
@vivekrnv please update the description with how this change was validated?

**Comment by vaibhavhd** [comment]
MSFT ADO: 25357782

**Comment by StormLiangMS** [comment]
@vivekrnv have you test with 202305?

**Comment by vivekrnv** [comment]
With this PR https://github.com/sonic-net/sonic-buildimage/pull/16116, this change is a good to have but not a must

**Pull request merged**